src/resource/crc32.rs

```diff
@@ -23,34 +31,37 @@ use crate::{ArklibError, Result};
     Deserialize,
     Serialize,
 )]
-pub struct ResourceId {
+pub struct ResourceIdCrc32 {
```

Comment:
Probably the types implementing the trait could be named `ResourceId`, too?

In the usage spot we always can use `crc32::ResourceId`. It seems to be more readable to me.